server/src/visitors/VisitorsUtils.ts

```diff
@@ -1,117 +1,104 @@
-import { Token } from "antlr4ts";
-import {
-  CompletionItemKind,
-} from "vscode-languageserver";
-import { YmlCompletionItemsProvider } from "../completion/YmlCompletionItemsProvider";
-import { IDefinitionLocation } from "../definitions";
-import { FieldContext, YmlIdContext } from "../grammar";
-import { connection } from "../server";
+import { Token } from 'antlr4ts';
+import { Location } from 'vscode-languageserver';
+
+import { FieldContext } from '../grammar';
+import { connection } from '../server';
+import { AbstractYmlObject } from '../yml-objects';
+
+/**
+ * Enrich an item by adding to it some informations like its type, its use scope, documentation, etc.
+ *
+ * @param completionItem The item to enrich.
+ * @param fields The attributes the item has to find from it more informations.
+ * @param sourceElementName The wrapper item. For an argument it is the function, for an attribute it is its class.
+ * @param baseType The type of the item, if it is not an information that can be found in `fields`
+ * @param scopeStartOffset The first position from which we can suggest this item.
+ * First character of a function's body, for example.
+ * Useful to avoid giving completion suggestions pertaining to symbols
+ * that are out of the current scope.
+ * @param scopeEndOffset The last position to which we can suggest this item.
+ * Last character of a function's body, for example.
+ * Useful to avoid giving completion suggestions pertaining to symbols
+ * that are out of the current scope.
+ */
+export function enrichCompletionItem(
```

Comment:
Any reason why not to turn this into a method in `AbstractYmlObject`? :thinking: